cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/transaction/coprocessor/CConfigurationCacheSupplier.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.transaction.coprocessor;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ * Supplies instances of {@link CConfigurationCache} implementations.
+ */
+public class CConfigurationCacheSupplier implements CacheSupplier<CConfigurationCache> {
```

Comment:
Is this so that we can derive from this abstract class for the two suppliers? Any ideas on how we can handle the static variables - instance, object, refCount?